basics/list_subdirectory.go

```diff
@@ -0,0 +1,20 @@
+/* This program usins filepath/walkfunc to list 
+all the files in a directory */
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+func main() {
+	dir := "/tmp"
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		fmt.Println("files :", path)
+		return nil
+	})
+	if err != nil {
```

Comment:
Check the example here https://play.golang.org/p/Uk5VngvDCb1
err should be returned immediately before anything